pkg/sql/create_table.go

```diff
@@ -1957,6 +1957,16 @@ func newTableDesc(
 		)
 	})
 
+	// See if any user defined types were used in this table.
+	if err := ret.ForeachPublicColumn(func(col *descpb.ColumnDescriptor) error {
```

Comment:
any reason to not re-use https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/create_table.go#L2205?

Reply:
`ColumnDef`s contain only `tree.ResolveableTypeReferences`. We only know the type and everything after we process the def into a column

Reply:
I'll just change it to operate on a column descriptor